### PR TITLE
MINIFICPP-1203 - upgrade linter version to 1.4.4 and fix relevant linter errors

### PR DESCRIPTION
**Pull request by hunyadi-dev**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by hunyadi-dev** [comment]
:exclamation: This PR depends on #769: ensuring that the `make linter` builds are clean with both versions of the linter.

Originally the linter was sourced from the google style-guide fork which is not very well maintained, this is now replaced with the script shipped by the cpplint/cpplint repository maintainers

File: libminifi/src/io/StreamFactory.cpp [hunk 1]
```diff
@@ -16,12 +16,13 @@
...
  */
 #include "io/StreamFactory.h"
+
 #include <atomic>
 #include <memory>
 #include <mutex>
 #include <string>
-#include <utils/GeneralUtils.h>
 
+#include <utils/GeneralUtils.h> // NOLINT
```
Comment (szaszm) [inline libminifi/src/io/StreamFactory.cpp:25]: why do we need NOLINT here?
Also, end of line comments should have 2 preceding spaces.

File: libminifi/test/archive-tests/FocusArchiveTests.cpp [hunk 2]
```diff
@@ -16,24 +16,23 @@
...
+#include <memory>
 #include <set>
 #include <string>
-#include <algorithm>
-#include <memory>
 #include <utility>
 
-#include "../TestBase.h"
+#include <archive.h> // NOLINT
+#include <archive_entry.h> // NOLINT
```
Comment (szaszm) [inline libminifi/test/archive-tests/FocusArchiveTests.cpp:28]: why do we need NOLINT here?

End of line comments should have 2 spaces before and 1 after the `//`.

File: libminifi/src/io/StreamFactory.cpp (hunk 1 above)
Comment (hunyadi-dev) [reply libminifi/src/io/StreamFactory.cpp:25]: This is incorrectly recognized as a C header. Also `NOLINT` comments do not need to be preceded by 2 spaces.

File: libminifi/test/archive-tests/FocusArchiveTests.cpp (hunk 2 above)
Comment (hunyadi-dev) [reply libminifi/test/archive-tests/FocusArchiveTests.cpp:28]: This is incorrectly recognized as a C header. Also `NOLINT` comments do not need to be preceded by 2 spaces.

**Comment by hunyadi-dev** [comment]
> I find the Google include ordering rules strange, but that's in CONTRIB.md, so I guess we're better off enforcing them.
> 
> Thanks for the upgrade!

Agreed on the rules being strange, I would much rather prefer the llvm standard too. I could patch the linter for changing the order, but that would mean that we would need to apply the same patch every time we want to upgrade the linter version (and that everyone developping a code with linter assistance would need to apply the same patch to their own linter). I don't think the change is worth the effort.

File: libminifi/src/io/StreamFactory.cpp (hunk 1 above)
Comment (szaszm) [reply libminifi/src/io/StreamFactory.cpp:25]: https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes

I think if we replaced the angle brackets with quotation marks, the warning would go away. Could you try this? If it doesn't work, them I'm fine with leaving the NOLINT comments of such cases in place.

File: libminifi/src/utils/OsUtils.cpp [hunk 3]
```diff
@@ -24,16 +24,18 @@
 #ifndef WIN32_LEAN_AND_MEAN
 #define WIN32_LEAN_AND_MEAN
 #endif
-#include <Windows.h>
 #include <sddl.h>
+#include <Windows.h>
```
Comment (szaszm) [inline libminifi/src/utils/OsUtils.cpp:28]: Judging from the appveyor output, the order of these includes matter. Not sure of the exact reason, but in general windows headers are a mess, so I'm not surprised. :(

See the end of the log here: https://ci.appveyor.com/project/ApacheSoftwareFoundation/nifi-minifi-cpp/builds/33004452

File: libminifi/test/tensorflow-tests/TensorFlowTests.cpp [hunk 4]
```diff
@@ -19,14 +19,14 @@
...
-#include <tensorflow/cc/framework/scope.h>
-#include <tensorflow/cc/ops/standard_ops.h>
-#include <processors/PutFile.h>
-#include <processors/GetFile.h>
-#include <processors/LogAttribute.h>
+#include <processors/GetFile.h> // NOLINT
+#include <processors/LogAttribute.h> // NOLINT
+#include <processors/PutFile.h> // NOLINT
+#include <tensorflow/cc/framework/scope.h> // NOLINT
+#include <tensorflow/cc/ops/standard_ops.h> // NOLINT
```
Comment (szaszm) [inline libminifi/test/tensorflow-tests/TensorFlowTests.cpp:26]: Regarding my previous suggestion: I suggest trying to convert our headers to use quotation marks and leave 3rd parties as is, even if it means that we need to use NOLINT.

Comment (hunyadi-dev) [reply libminifi/test/tensorflow-tests/TensorFlowTests.cpp:26]: Will do that in a separate PR, right? I don't want to rebase my linter PR again.

File: libminifi/src/utils/OsUtils.cpp (hunk 3 above)
Comment (hunyadi-dev) [reply libminifi/src/utils/OsUtils.cpp]: Corrected.

File: libminifi/test/tensorflow-tests/TensorFlowTests.cpp (hunk 4 above)
Comment (szaszm) [reply libminifi/test/tensorflow-tests/TensorFlowTests.cpp:26]: My initial idea was to do it here, but you're right, it's better not to mix in this.

**Comment by arpadboda** [comment]
Merged in 29460fa1a035096b251a7f2a323ddffcec2c4210

**Pull request closed without merging**